stream_alert/apps/_apps/intercom.py

```diff
@@ -0,0 +1,103 @@
+import time
+import re
+
+import requests
+
+from . import AppIntegration, StreamAlertApp, get_logger
+
+
+LOGGER = get_logger(__name__)
+
+
+@StreamAlertApp
+class IntercomApp(AppIntegration):
+    """Intercom StreamAlert app"""
+    _INTERCOM_LOGS_URL = 'https://api.intercom.io/admins/activity_logs'
+
+    def __init__(self, event, config):
+        super(IntercomApp, self).__init__(event, config)
+        self._next_page = None
+
+    @classmethod
+    def _type(cls):
+        return 'admin_activity_logs'
+
+    @classmethod
+    def service(cls):
+        return 'intercom'
+
+    @classmethod
+    def _required_auth_info(cls):
+        return {
+            'token': {
+                'description': 'the OAuth token for this Intercom app',
+                'format': re.compile(r'.*')
+            }
+        }
+
+    def _sleep_seconds(self):
+        """Return the number of seconds this polling function should sleep for
+        between requests to avoid failed requests. Intercom API allows for a default of 500 requests
+        every minute, distributed over 10 seconds periods. This means for a default rate limit of
+        500 per minute, you can send a maximum of 83 operations per 10 second period.
+
+        Resource(s):
+            https://developers.intercom.com/intercom-api-reference/reference#rate-limiting
+
+        Returns:
+            int: Number of seconds that this function should sleep for between requests
+        """
+        return 10
```

Comment:
given your docstring description of their rate limiting, it doesn't sound like this should actually be necessary 🤔 . I don't think the lambda will be able to do more than 83 requests a minute, so sleeping probably isn't necessary